frontend/components/AdminItemsTable.js

```diff
@@ -1,18 +1,51 @@
-import React from "react";
+import React, { useState } from "react";
 import { mutate } from "swr";
 import PropTypes from "prop-types";
 import { itemType } from "../types";
 import { fetchIt } from "../utils";
 import Button from "./Button";
+import Modal from "./Modal";
 
-const handleDelete = id => {
+/*const handleDelete = id => {
   mutate("/items", async items => {
     await fetchIt(`/item/${id}`, { method: "DELETE" });
     return items.filter(item => item.id === id);
   });
+  setIsConfirmingDelete(false);
 };
 
+const handleClose = () => {
+  setIsConfirmingDelete(false);
+};
+
+const confirmDelete = id => {
+  setDeleteId(id);
+  setIsConfirmingDelete(true);
+};*/
+
 const AdminItemsTable = ({ items }) => {
+  const handleDelete = id => {
+    mutate("/items", async items => {
+      await fetchIt(`/item/${id}`, { method: "DELETE" });
+      return items.filter(item => item.id === id);
+    });
+    setIsConfirmingDelete(false);
+  };
+
+  const handleClose = () => {
+    setIsConfirmingDelete(false);
+  };
+
+  const confirmDelete = (id, itemName) => {
+    setDeleteId(id);
+    setDeleteItemName(itemName);
+    setIsConfirmingDelete(true);
+  };
+
+  const [isConfirmingDelete, setIsConfirmingDelete] = useState(false);
+  const [deleteId, setDeleteId] = useState(false);
+  const [deleteItemName, setDeleteItemName] = useState(false);
```

Comment:
Can the state declarations be moved above all the functions?

Reply:
Yes